### ISSUE TITLE
Superpmi collection for libraries_tests broken for all OS/arch

### ISSUE THREAD

**Issue by kunalspathak** [dotnet/runtime#56836]
From the logs, I see following errors during collection.

```
'C:\h\w\932D0886\p\superpmi\corerun.exe C:\h\w\932D0886\p\superpmi\pmi.dll DRIVEALL C:\h\w\932D0886\w\AE390948\u\binaries\Microsoft.Build.dll': Error return code: 3762504530
============== Contents of C:\h\w\932D0886\t\tmpic4qiz4x\pmi_C__h_w_932D0886_w_AE390948_u_binaries_Microsoft_Build_dll_nbefdkqo.stderr
Unhandled exception. System.BadImageFormatException: Could not load file or assembly 'System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (0x80131058)
File name: 'System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' ---> System.BadImageFormatException: Cannot load a reference assembly for execution.

```

It worked fine until 5bcf2c67107a9790b4638d3414d728916327230c and was broken when we did collection for e8be7f26.

```
git lg 5bcf2c..e8be7f26
[ e8be7f26cb1 ] 2021-07-18 17:18 tal.alon.. Bugfix: System.Xml.Serialization: Compiler.GetTempAssemblyName is not deterministic under .NET Core (#46499)
[ 7eb749c8b78 ] 2021-07-18 06:06 prgovi@m.. Eliminate backtracking in the interpreter for patterns with .*  (#51508)
[ fe49e55c09c ] 2021-07-18 05:45 josalem@.. set CLOEXEC on diagnostic server FDs (#55850)
[ 6b6310b1c96 ] 2021-07-18 04:05 jbarton@.. Reduce the nullability bypasses in CryptoPool.Return (#55782)
[ 44df0a37632 ] 2021-07-18 07:03 42748379.. [main] Update dependencies from dnceng/internal/dotnet-optimization dotnet/arcade dotnet/icu dotnet/roslyn-analyzers (#55812)
[ 1d0ee125ee1 ] 2021-07-18 07:01 stoub@mi.. Fix strings in ApplicationId.ToString (#55879)
[ d66305978cb ] 2021-07-18 06:39 stoub@mi.. Update Roslyn to 4.0.0-3.21367.2 (#55878)
[ 26f5cc54f37 ] 2021-07-18 19:31 gengesa@.. Add VersionCheck (#55763)
[ 96ce6b35359 ] 2021-07-18 03:12 tweinfur.. enable two quic tests (#55880)
[ 3906d0df62f ] 2021-07-17 23:42 Kunal.Pa.. Consider CALLFINALLY block as pred of finally block during assertion props (#55674)
```

**Comment by JulieLeeMSFT** [comment]
@kunalspathak setting this as .NET6 for now.

**Comment by kunalspathak** [comment]
I spent time on this today but was not able to get to the bottom of this. I noticed that the `System.Runtime.dll` that we send as payload has different size in failing vs. passing pipeline. When `libraries_tests` worked, the size was approx. 32KB and currently it is 655KB. It happened around the time when https://github.com/dotnet/runtime/pull/55678 went in. Once https://github.com/dotnet/runtime/pull/57084 is merged, I will do a collection run on commit of #55678  and the one before.
